### PR TITLE
Added functionality to read gpg keys from the keys directory

### PR DESCRIPTION
**Pull request by Yantrio**
This PR also adds the opentofu gpg keys as a sample to ensure this works.

### CODE REVIEW

File: src/internal/v1api/providers.go [hunk 1]
```diff
@@ -69,6 +74,16 @@ func (p ProviderGenerator) VersionListing() ProviderVersionListingResponse {
...
 	versionDetails := make(map[string]ProviderVersionDetails)
 
+	keyCollection := gpg.KeyCollection{
+		Namespace: p.Provider.EffectiveNamespace(),
+		Directory: p.KeyLocation,
+	}
+
+	keys, err := keyCollection.ListKeys()
+	if err != nil {
+		p.log.Error("Failed to list keys", slog.Any("err", err))
```
Comment (cam72cam) [inline src/internal/v1api/providers.go:84]: Do we want to return the error?

**Review by cam72cam — APPROVED** [review]
Overall looks good to me, just not sure if we want to kill a provider if keys list does not work.

File: src/internal/v1api/providers.go (hunk 1 above)
Comment (Yantrio) [reply src/internal/v1api/providers.go:84]: we probably should. I'll add this tomorrow morning. Good catch!

**Comment by Yantrio** [comment]
I've added this, If we cannot read the gpg key for some reason, i feel like we shouldn't update the rendered v1 responses at all. I would rather have outdated than broken responses.

File: keys/o/opentofu/20230928.asc [hunk 2]
```diff
@@ -0,0 +1,29 @@
+-----BEGIN PGP PUBLIC KEY BLOCK-----
```
Comment (cube2222) [inline keys/o/opentofu/20230928.asc]: I think we can get rid of this key?

Comment (Yantrio) [reply keys/o/opentofu/20230928.asc]: Yup, we can. I'll remove it.

File: src/internal/gpg/keycollection.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	files, err := os.ReadDir(location)
+	if err != nil {
+		return nil, err
+	}
+
+	keys := make([]Key, len(files))
+	for i, file := range files {
+		key, err := buildKey(filepath.Join(location, file.Name()))
+		if err != nil {
+			return nil, err
```
Comment (cube2222) [inline src/internal/gpg/keycollection.go]: error wrap maybe for the given directory?

**Review by cube2222 — APPROVED** [review]

**Pull request merged**